sql/sql_udf.cc

```diff
@@ -206,6 +206,15 @@ void udf_init()
     DBUG_PRINT("info",("init udf record"));
     LEX_CSTRING name;
     name.str=get_field(&mem, table->field[0]);
+
+    // Check the name.str is NULL or not.
+    if (name.str == nullptr)
```

Comment:
The name in the mysql.func table is declared `NOT NULL` so this isn't the case that is hit when `insert into mysql.func(ret) values(1);` is performed.

name will be an empty string (or easier `name.length=0`)